crates/forge/src/test_execution_syscall_handler.rs

```diff
@@ -258,39 +258,47 @@ impl TestExecutionSyscallHandler<'_, '_> {
                 Ok(())
             }
             "start_warp" => {
-                let contract_address = inputs[0].to_contract_address();
-                let value = inputs[1].clone();
+                // The last element in `inputs` should be the timestamp in all cases
+                let warp_timestamp = inputs.last().unwrap().clone();
+
+                let target = deserialize_cheat_target(&inputs[..inputs.len() - 1]);
+
                 self.contract_execution_syscall_handler
                     .cheatable_syscall_handler
                     .cheatnet_state
-                    .start_warp(contract_address, value);
+                    .start_warp(target, warp_timestamp);
+
                 Ok(())
             }
             "stop_warp" => {
-                let contract_address = inputs[0].to_contract_address();
+                let target = deserialize_cheat_target(&inputs);
+
                 self.contract_execution_syscall_handler
                     .cheatable_syscall_handler
                     .cheatnet_state
-                    .stop_warp(contract_address);
+                    .stop_warp(target);
+
                 Ok(())
             }
             "start_prank" => {
-                let contract_address = inputs[0].to_contract_address();
-                let caller_address = inputs[1].to_contract_address();
+                // The last element in `inputs` should be the contract address in all cases
+                let caller_address = inputs.last().unwrap().to_contract_address();
+
+                let target = deserialize_cheat_target(&inputs[..inputs.len() - 1]);
```

Comment:
Let's change the order of these two (we should extract them the same way they were passed)